core/src/main/java/org/apache/iceberg/avro/ValueReaders.java

```diff
@@ -591,6 +591,8 @@ protected StructReader(List<ValueReader<?>> readers, Schema schema) {
         if (AvroSchemaUtil.getFieldId(field) == MetadataColumns.ROW_POSITION.fieldId()) {
           // track where the _pos field is located for setRowPositionSupplier
           this.posField = pos;
+        } else if (AvroSchemaUtil.getFieldId(field) == MetadataColumns.IS_DELETED.fieldId()) {
+          this.readers[pos] = new ConstantReader<>(false);
```

Comment:
Can you convert this to use positions and constants as well?

Reply:
Oops, my brain was not working well this early morning... Let me take a coffee first.